lib/new-utils.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+const Long = require("long");
+
+// maxInt value is based on how does Long splits values between internal high and low fields.
+const maxInt = BigInt(0x100000000);
+const minusOne = BigInt(-1);
+
+/**
+ * Converts from bigint provided by napi into Long type used by the datastax library
+ * BigInt is the way napi handles values too big for js Number type,
+ * while Long is the way datastax parts of the code handle 64-bit integers.
+ * @param {bigint} from
+ * @returns {Long}
+ */
+function bigintToLong(from) {
+  let lo = from % maxInt;
+  let hi = from / maxInt;
+  if (lo < 0) hi += minusOne;
+  return Long.fromValue({ low: Number(lo), high: Number(hi), unsigned: false });
+}
```

Comment:
We're using `Long` only because Datastax driver exposes it in its API, right? In the office we concluded that `BigInt` is available for us in the minimum supported NodeJS version, that's why I'm asking if Long is needed at all.

Reply:
Not only because of that. Currently datastax in it's code uses Long for all internal i64 calculations, and changing that would require not insignificant amount of effort, thanks to functions like this: https://github.com/scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver/blob/6cc023e07142e8ea6c8d8a7d49b52a40d3784a75/lib/types/duration.js#L84-L87
that assume the long is provided. Also the Long has different semantic for even basic operations like shown here:
https://github.com/scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver/blob/6cc023e07142e8ea6c8d8a7d49b52a40d3784a75/lib/types/duration.js#L119-L120

Reply:
Understood. Then please create an issue about possible future replacement of Long with BigInt in the whole codebase, and how much effort it's going to be and why.